src/adam_bashforth.jl

```diff
@@ -1,57 +1,58 @@
-module Tmp
 ####################################
 # Explicit Adam-Bashforth solvers
 ####################################
 # (Hairer & Wanner 1996, Vol I, p.357-358
-include("ODE.jl")
-
-##Fixed step, fixed order, multistep explicit method
-
-function ode_ab(F,y0, tspan,order ::Integer)
-    if !(2 <= order <= 4)
-        error("Currently only orders 2,3, and 4 are implemented for Adam-Bashforth method")
-    end
-
-    stepsize = diff(tspan)
-    y = Array(typeof(y0), length(tspan))
-    y[1] = y0
-
-    ##Use Runge-Kunta for initial points necessary to base Adam Basforth off of, if initial values not given
-    t[i:N], y[i:N] = ode_ms(F,y0, tspan[1:N], order)
-
-    ##Use Adam Basforth method for subsequent steps
-    Integrator = AdamBashforth{order}
-    @inbounds for i = N+1:length(tspan)
-        y = step(Integrator,f,y,t,stepsize,i-1) ::Float64
-        y[i] = y
-        t = t + stepsize
-    end
-    ##Return y values
-    tspan, y
-end
+#Begin AdamBash module
+module AdamBash
 
-##Implementation of the Adam Steps for order 2, 3 and 4
+using Polynomials;
+using Compat;
+include("ODE.jl")
 
+##Implementation of the Adam Steps for order 2, 3 and 4 using recursive formualae from Hairer
 abstract Integrator
-immutable AdamBashforthStep{N} end
+immutable AdamBashforth{N} end
 
-@inline function adam_step(::Union{AdamBashforthStep{2},Type{AdamBashforthStep{2}}}, f, y, t, stepsize)
+@inline function adam_step(::Union{AdamBashforth{2},Type{AdamBashforth{2}}}, f, y, t, stepsize, i)
     ynext = y[i] + stepsize/2*(3*f(t[i], y[i])
     - 1*f(t[i-1],y[i-1]))
 end
-
-@inline function step(::Union{AdamBashforthStep{3},Type{AdamBashforthStep{3}}}, f, y, t, stepsize)
+@inline function adam_step(::Union{AdamBashforth{3},Type{AdamBashforth{3}}}, f, y, t, stepsize, i)
     ynext = y[i] + stepsize/12*(23*f(t[i], y[i])
     - 16*f(t[i-1],y[i-1])
     + 5*f(t[i-2],y[i-2]))
 end
-
-
-@inline function step(::Union{AdamBashforthStep{4},Type{AdamBashforthStep{4}}}, f, y, t, stepsize)
+@inline function adam_step(::Union{AdamBashforth{4},Type{AdamBashforth{4}}}, f, y, t, stepsize, i)
     ynext = y[i] + stepsize/24*(55*f(t[i], y[i])
     - 59*f(t[i-1],y[i-1])
     + 37*f(t[i-2],y[i-2])
     - 9*f(t[i-3],y[i-3]))
 end
 
+##Function: ode_abe(F,y0, tspan, order(optional))
+##order is set to 4 by default
+##Adam Bashforth is a fixed step, fixed order, multistep explicit method
+function ode_ab(F::Function,y0, tspan,order=4 ::Integer)
+    if !(2 <= order <= 4)
+        error("Currently only orders 2,3, and 4 are implemented for Adam-Bashforth method")
+    end
+
+    h = diff(tspan)
+    yvals = Array(typeof(y0), length(tspan))
+    yvals[1] = y0
+
+    ##Use Runge-Kunta for initial points necessary to base Adam Basforth off of, if initial values not given
+    tint, yint= ODE.ode_ms(F,y0, tspan[1:order], order)
+
+    for i = 1 : order
+        yvals[i] = yint[i]
+    end
+    ##Use Adam Basforth method for subsequent steps
+    Integrator = AdamBashforth{order}
```

Comment:
`integrator = AdamBashforth{order}()`, the `()` makes an instance, usually preferred.